lib/rule-tester/rule-tester.js

```diff
@@ -521,6 +521,21 @@ class RuleTester {
             ].concat(scenarioErrors).join("\n"));
         }
 
+        if (typeof rule === "function") {
+            util.deprecate(
+                () => {},
+                `"${ruleName}" rule is using the deprecated function-style format and will stop working in ESLint v9. Please use object-style format: https://eslint.org/docs/developer-guide/working-with-rules`,
+                "DEP_ESLINT_LEGACY_RULE_API"
```

Comment:
According to `util.deprecate` docs, if we specify a deprecation code, then the warning will be emitted only once for that code, while I think we want the warning to be emitted for each rule that is using the function-style format.